### PR TITLE
[Feat] Search Box Component

### PR DESCRIPTION
**Pull request by r-aya**
## Notion ticket link

[Search Function: Search Box Component](https://www.notion.so/uwblueprintexecs/Search-Function-Search-Box-Component-107a6b6ffc8a47a18e19b5e07617bfea)

## Implementation description
* Created Search Box component using Chakra's input component and icons

## Steps to test
1. Type into input field
2. Click Search to ensure cursor changes
3. Click left icon to ensure cursor doesn't change
4. Check styling for focus and hover

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have run docker-compose up and my project compiled
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by hyunzeekim — COMMENTED** [review]
code looks good! one minor thing when testing: i noticed that if the search bar is filled with words until the end, then the words behind the "Search" button is cut off and can't be seen even if you try to highlight all the way until the end to scroll.

File: frontend/src/components/pages/SearchBox.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import {
+  Button,
+  Input,
+  InputGroup,
+  InputLeftElement,
+  InputRightElement,
+} from "@chakra-ui/react";
+import React from "react";
+
+const SearchBox = (): React.ReactElement => {
```
Comment (susheelk) [inline frontend/src/components/pages/SearchBox.tsx:11]: Hey, so this ticket also involves keeping track of the search text as state. I think the previous dev forgot to implement this. Could you:
1. Take a function called `setSearchText` as a prop
2. Upon click of the search button, call this function with the entered search text

Thanks!

**Review by susheelk — CHANGES_REQUESTED** [review]
Code looks good! There's just a requirement that I forgot about lol. Also the deploy job's failing, could you look into that?

**Review by susheelk — APPROVED** [review]
LGTM! Nice work! :tada:

**Pull request merged**